Meden-Agan-27/travel-worms
# Friendship functionality

What I did:
built friendship functionality where you can send friend requests to people, and they can accept, you 
can also accept or decline friend requests coming in to you.

Made bookshelves of friends visible on their profile if you are friends

Added a Find Users button to the navbar drop down.

How to test:
-Log in
-Go to the nav bar, Find Users
-search for someone's username (case sensitive for now, so it has to be unique)
-press the befriend button
-check you can't see the bookshelves on their profile
-log out
-log in on the person you just sent a friend request to
-go to My Profile
-check that "Pending Friend Requests" appears with the option to accept
- click accept
- click the name to go to their profile
- see if the bookshelves appear and you can go into them
- you should not be able to edit the reading status or delete items from the bookshelf.

There are some small layout things wrong but we should be able to fix that tomorrow.
